src/rubicon/objc/api.py

```diff
@@ -897,7 +967,8 @@ def __del__(self):
         # Autorelease our reference on garbage collection of the Python wrapper. We use
         # autorelease instead of release to allow ObjC to take ownership of an object when
         # it is returned from a factory method.
-        send_message(self, "autorelease", restype=objc_id, argtypes=[])
+        if send_message and objc_id:
```

Comment:
How could `send_message` or `objc_id` ever be unavailable?

Reply:
During teardown of the Python interpreter, for example after calling `exit()`, see warning in the  [\_\_del\_\_](https://docs.python.org/3/reference/datamodel.html#object.__del__) documentation. Maybe checking for `None` is not the best way to handle this since it is (1) not race free and (2) does not handle if those objects are deleted instead of set to None.

Catching `NameError` (object was deleted) and `TypeError` (object was set to None) may be a safer in this regard.

Reply:
FYI, I am actually seeing those errors printed to stderr in Python 3.13 on exit if there is no special handling.